### ISSUE TITLE
Issues developing on modern Ubuntu

### ISSUE THREAD

**Issue by marthinwurer** [magefree/mage#6458]
Modern Ubuntu distributions no longer ship with Java 8 compatible JavaFX builds. It's pretty hard to acquire them for newer ubuntu distros. I looked for a while, and the only way to do it was to use the Oracle distribution that had really restrictive licensing. So, I had to build it from source. That was a whole [adventure in and of itself](https://wiki.openjdk.java.net/display/OpenJFX/Building+OpenJFX+8u). I was finally able to get xMage itself to compile.

So, that being said, I think there's gonna be more issues in the future with the exact java version that we're using. Oracle is litigious, so if we're using their distribution then we should move away from it ASAP. OpenJDK doesn't have some of those built-in libraries, so we'll have to add them to the maven packages or compile them from source, which is a huge pain. The JavaFX builds that are on maven are only for Java 11+, so we'd have to upgrade the java version too.

**Comment by timhae** [comment]
Is this a lot more complicated than replacing all occurences of 1.8 in `<source>` and `<target>` sections in `pom.xml`s with 11 and adding
```
<dependency>
    <groupId>org.openjfx</groupId>
    <artifactId>javafx-base</artifactId>
    <version>15-ea+4</version>
</dependency>
```
to the Mage.Client  pom? I wanted to contribute to the project but so far I was unable to build it on linux, this seems to work quite alright and I am finally able to build :) there are some exceptions in the deck editor though but if someone would like to help me out every now and then I would try to get this done. Or do you think that might not be suitable as a first issue?

**Comment by ThomasArb** [comment]
I had the same issue here.
What I did is : 
- Install openJDK 1.8
- Change the code of `WhatsNewDialog.java` by that : 
```java
public class WhatsNewDialog extends MageDialog {

    public WhatsNewDialog() {
    }

    public void checkUpdatesAndShow(boolean toto) {
        return;
    }
}
```

Now it run's without any problem.

**Comment by marthinwurer** [comment]
I mean, that's not really a solution. You can remove all the errors in your code by deleting the code, but it's still not going to have the desired functionality.

**Comment by ThomasArb** [comment]
Of course, I totaly agree with you. But I just share what I did in order to get xmage to work quickly to just do one or two text corrections.

**Comment by allentiak** [comment]
The only way I know of building the project under GNU/Linux is using either Oracle's JDK8 or OpenJDK8 + OpenJFX8. OpenJFX8 is supported by all distros. The problem is OpenJFX8.

Old Debian releases provide OpenJFX8 builds[1]. Although I have not found any Ubuntu OpenJFX8 packages, perhaps you could use Debian's? Probably other distros provide OpenJFX8 builds as well. Another option is 
[1]: https://packages.debian.org/search?keywords=openjfx

Shall we continue the discussion in #6197?
